moto/guardduty/responses.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import unicode_literals
+from moto.core.responses import BaseResponse
+from .models import guardduty_backends
+import json
+
+
+class GuardDutyResponse(BaseResponse):
+    SERVICE_NAME = "guardduty"
+
+    @property
+    def guardduty_backend(self):
+        return guardduty_backends[self.region]
+
+    # @property
+    # def request_params(self):
+    #     try:
+    #         return json.loads(self.body)
+    #     except ValueError:
+    #         return {}
+
+    # def _get_param(self, param, default=None):
+    #     return self.request_params.get(param,default)
+
+    def detector(self, request, full_url, headers):
+        self.setup_class(request, full_url, headers)
+        if request.method == "POST":
+            return self.create_detector()
+        elif request.method == "GET":
+            return self.list_detectors()
+        else:
+            return 404, {}, ""
+
+    def create_detector(self):
+        enable = self._get_param("enable")
+        client_token = self._get_param("clientToken")
+        finding_publishing_frequency = self._get_param("findingPublishingFrequency")
+        data_sources = self._get_param("dataSources")
+        tags = self._get_param("tags")
+
+        detector_id = self.guardduty_backend.create_detector(
+            enable, client_token, finding_publishing_frequency, data_sources, tags
+        )
+
+        return 200, {}, json.dumps(dict(detectorId=detector_id))
```

Comment:
With my change, each method now needs to return a tuple in the form of (HTTP Status Code, Headers, Body (as a string)). Bit more code, but it does make it easier to fine-tune the response if required